### PR TITLE
Fix/8541: Certificate validity: Spinning the country selection wheel and closing the section prevents UI update

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
If a user spins the wheel, and collapse the section, the row selection should be prevented.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8541

## Screenshots

https://user-images.githubusercontent.com/22373291/209806024-01120f8f-3a33-4093-b1ae-25119c0b72a6.mp4

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/CountrySelectionCell.swift [hunk 1]
```diff
@@ -147,6 +153,10 @@ final class CountrySelectionCell: UITableViewCell, UIPickerViewDelegate, UIPicke
 	}
 
 	func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+		guard delegate?.isAllowedCountrySelection ?? false else {
+			// If an user spins the wheel, and collapse their while the cell, the row selection should be prevented.
```
Comment (puneetmahali1) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/CountrySelectionCell.swift]:
```suggestion
			// If an user spins the wheel and collapse it then the row selection should be prevented.
```

**Review by puneetmahali1 — APPROVED** [review]
LGTM! Just a minor change

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/CountrySelectionCell.swift [hunk 2]
```diff
@@ -147,6 +153,10 @@ final class CountrySelectionCell: UITableViewCell, UIPickerViewDelegate, UIPicke
 	}
 
 	func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+		guard delegate?.isAllowedCountrySelection ?? false else {
+			// If an user spins the wheel and collapse it then the row selection should be prevented.
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/CountrySelectionCell.swift]:
```suggestion
			// If a user spins the wheel and collapse it then the row selection should be prevented.
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/__tests__/CountrySelectionCellTests.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import XCTest
+@testable import ENA
+
+final class CountrySelectionCellTests: XCTestCase {
+
+    func testPickerViewDidSelectRow_delegate_isAllowedCountrySelection_true_selectCountry() throws {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/__tests__/CountrySelectionCellTests.swift:10]: 👍

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**